### DISCUSSION TITLE
Do rows without columns make sense?

### DISCUSSION THREAD

**Discussion by jonmor51** [quasarframework/quasar#13168] [General - Components / Directives / etc] [answered]
Years of using Bootstrap (mostly badly) have trained me always to use `<div class="col-xx"...>` within every `<div class="row"..>`, a practice I've carried over into Quasar. But is this, in fact, necessary? I note that I can do something like:

```
<div class='row'>
 <q-input .../>
 <q-input .../>
 <q-input .../>
</div>
```

and get a bunch of inputs on the same line. Is this bad practice, though?

**Comment by MilosPaunovic** [comment, marked as answer]
Depends on what you're trying to achieve. Best would be to go through the pages about flexbox on the docs [https://quasar.dev/layout/grid/introduction-to-flexbox](https://quasar.dev/layout/grid/introduction-to-flexbox).

**Reply by jonmor51** [reply]
Thanks, but I've read the documentation. I was hoping for a bit more personal input from people who are using Quasar regularly.

**Comment by metalsadman** [comment]
like what Milos said, it depends on you if you need classes that works along with it not just `col`s, ie. wrapping, vertical or horizontal alignment etc... if you don't need any of that then i don't see a need of using it.

**Reply by jonmor51** [reply]
So what does `class = 'row'` do, then? Does it just prevent the contained inputs etc from being block-level, as they otherwise would be?
